app/lib/components/term.js

```diff
@@ -181,16 +218,12 @@ export default class Term extends Component {
     }
 
     if (this.props.cursorColor !== nextProps.cursorColor) {
-      this.term.prefs_.set('cursor-color', nextProps.cursorColor);
+      this.setCssVariable('--cursor-color', nextProps.cursorColor);
     }
 
     if (this.props.colors !== nextProps.colors) {
       this.term.prefs_.set('color-palette-overrides', nextProps.colors);
     }
-
-    if (this.props.customCSS !== nextProps.customCSS) {
-      this.term.prefs_.set('user-css', this.getStylesheet(nextProps.customCSS));
```

Comment:
same way, I just moved this block a couple lines above to avoid a race condition if both customCSS and one of the colors changed simultaneously